Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -526,6 +526,12 @@ public final class PackageBuilder {
                 return path
             }
 
+            // If there is only one target defined, it may be allowed to occupy the
+            // entire predefined target directory.
+            if self.manifest.targets.count == 1 {
```

Comment:
I think we should also check for the tools-version here to avoid people creating packages that are unexpectedly not backwards-compatible.

Reply:
Sounds good.

Reply:
doesn't this change _force_ it to occupy `./Sources`? I think it should _allow_ it not _force_ it. ie. we should also check if `./Sources/<target>` exist first and only if it does not fall back to `./Sources`

Reply:
Ah yes, this check should move below checking if `path` exists.

Reply:
Er, wait. No, it still works. This still happens first:

```
// Return the path if the predefined directory contains it.
if predefinedDir.contents.contains(target.name) {
    return path
}
```

Reply:
Okay, added logic for this and tests for 5.9 and the default tools version.

Reply:
ah nice!